src/integration/resources/inprocess/dbnode.go

```diff
@@ -0,0 +1,309 @@
+		Zone:           "embedded",
+		Weight:         1024,
+		Address:        "0.0.0.0",
+		Port:           uint32(port),
+	}, nil
+}
+
+func (d *dbNode) Health() (*rpc.NodeHealthResult_, error) {
+	return d.tchanClient.TChannelClientHealth(defaultRPCTimeout)
+}
+
+func (d *dbNode) WaitForBootstrap() error {
+	return retry(func() error {
+		health, err := d.Health()
+		if err != nil {
+			return err
+		}
+
+		if !health.GetBootstrapped() {
+			err = fmt.Errorf("not bootstrapped")
+			d.logger.Error("node not bootstrapped", zap.Error(err))
+			return err
+		}
+
+		return nil
+	})
+}
+
+func (d *dbNode) WritePoint(req *rpc.WriteRequest) error {
+	return d.tchanClient.TChannelClientWrite(defaultRPCTimeout, req)
+}
+
+func (d *dbNode) WriteTaggedPoint(req *rpc.WriteTaggedRequest) error {
+	return d.tchanClient.TChannelClientWriteTagged(defaultRPCTimeout, req)
+}
+
+func (d *dbNode) AggregateTiles(req *rpc.AggregateTilesRequest) (int64, error) {
+	res, err := d.tchanClient.TChannelClientAggregateTiles(defaultRPCTimeout, req)
+	if err != nil {
+		return 0, err
+	}
+
+	return res.ProcessedTileCount, nil
+}
+
+func (d *dbNode) Fetch(req *rpc.FetchRequest) (*rpc.FetchResult_, error) {
+	return d.tchanClient.TChannelClientFetch(defaultRPCTimeout, req)
+}
+
+func (d *dbNode) FetchTagged(req *rpc.FetchTaggedRequest) (*rpc.FetchTaggedResult_, error) {
+	return d.tchanClient.TChannelClientFetchTagged(defaultRPCTimeout, req)
+}
+
+func (d *dbNode) Exec(commands ...string) (string, error) {
+	//nolint:gosec
+	cmd := exec.Command(commands[0], commands[1:]...)
+
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	if err := cmd.Run(); err != nil {
+		return "", err
+	}
+
+	return out.String(), nil
+}
+
+func (d *dbNode) GoalStateExec(verifier resources.GoalStateVerifier, commands ...string) error {
+	return retry(func() error {
+		if err := verifier(d.Exec(commands...)); err != nil {
+			d.logger.Info("goal state verification failed. retrying")
+			return err
+		}
+		return nil
+	})
+}
+
+func (d *dbNode) Restart() error {
+	if err := d.Close(); err != nil {
+		return err
+	}
+
+	d.start()
+
+	return nil
+}
+
+func (d *dbNode) Close() error {
+	defer func() {
+		for _, dir := range d.tmpDirs {
+			if err := os.RemoveAll(dir); err != nil {
+				d.logger.Error("error removing temp directory", zap.String("dir", dir), zap.Error(err))
+			}
+		}
+	}()
+
+	for i := 0; i < d.cfg.Components(); i++ {
+		select {
+		case d.interruptCh <- errors.New("in-process node being shut down"):
+		case <-time.After(5 * time.Second):
+			return errors.New("timeout sending interrupt. closing without graceful shutdown")
+		}
+	}
+
+	for i := 0; i < d.cfg.Components(); i++ {
+		select {
+		case <-d.shutdownCh:
+		case <-time.After(1 * time.Minute):
+			return errors.New("timeout waiting for shutdown notification. server closing may" +
+				" not be completely graceful")
+		}
+	}
+
+	return nil
+}
+
+func updateDBNodePorts(cfg config.Configuration) (config.Configuration, error) {
```

Comment:
Note, this is not the complete list of addresses we could watch for based on the config, but one could see how we would extend this to add support